symphony-bdk-legacy/symphony-bdk-bot-sdk-java/src/test/java/com/symphony/bdk/bot/sdk/extapp/authentication/AppAuthControllerTest.java

```diff
@@ -0,0 +1,127 @@
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class AppAuthControllerTest {
+
+  private ExtensionAppAuthClient extAppAuthClient;
+  private ConfigClient configClient;
+  private AppAuthController appAuthController;
+  private AppToken appToken;
+  private JwtInfo jwtInfo;
+  private HttpServletRequest httpServletRequest;
+  private HttpServletResponse httpServletResponse;
+
+  @Before
+  public void init(){
+      this.configClient = mock(ConfigClient.class);
+      assertNotNull(this.configClient);
+      Mockito.when(this.configClient.getExtAppAuthPath()).thenReturn("extAppAuthPath");
+
+      this.extAppAuthClient = mock(ExtensionAppAuthClient.class);
+      assertNotNull(this.extAppAuthClient);
+
+      this.appToken = mock(AppToken.class);
+      Mockito.when(this.appToken.getAppToken()).thenReturn("appToken");
+      Mockito.when(this.appToken.getSymphonyToken()).thenReturn("SymphonyToken");
+
+      this.jwtInfo = mock(JwtInfo.class);
+      Mockito.when(this.jwtInfo.getJwt()).thenReturn("JWT");
+
+      this.appAuthController = new AppAuthController(this.extAppAuthClient, this.configClient, true);
+      assertNotNull(this.appAuthController);
+
+      this.httpServletRequest = mock(HttpServletRequest.class);
+      Mockito.when(this.httpServletRequest.getContextPath()).thenReturn("JWT");
+
+      this.httpServletResponse = mock(HttpServletResponse.class);
+      Mockito.doNothing().when(this.httpServletResponse).addCookie(any(Cookie.class));
+  }
+
+  @Test
+  public void testAuthenticate() {
+    final AppInfo appInfo = new AppInfo();
+    ResponseEntity responseEntity;
+
+    // Initialize configClient.extAppId
+    Mockito.when(this.configClient.getExtAppId()).thenReturn("56789");
+
+    // AppId = null
+    appInfo.setAppId(null);
+    responseEntity = this.appAuthController.authenticate(appInfo);
+    assertEquals(ResponseEntity.badRequest().build(), responseEntity);
+
+    // AppId != configClient.extAppId
+    appInfo.setAppId(this.configClient.getExtAppId() + "Test1");
+    responseEntity = this.appAuthController.authenticate(appInfo);
+    assertEquals(ResponseEntity.status(HttpStatus.UNAUTHORIZED).build(), responseEntity);
+
+    // AppId = configClient.extAppId
+    appInfo.setAppId(this.configClient.getExtAppId());
+    responseEntity = this.appAuthController.authenticate(appInfo);
+    AuthenticateResponse authenticateResponse = null;
+    try {
+      assertNotNull(appInfo.getAppId());
+      authenticateResponse = this.extAppAuthClient.appAuthenticate(appInfo.getAppId());
+      assertEquals(ResponseEntity.ok(authenticateResponse), responseEntity);
+    } catch (final AppAuthenticateException aae){
```

Comment:
Separate these tests: one test for each exception type.

Reply:
I can't separate them because the method `appAuthenticate` needs the 2 exceptions. If I remove one catch I have an error